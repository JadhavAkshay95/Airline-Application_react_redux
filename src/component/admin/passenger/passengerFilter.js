import React from 'react';
import { Container, Row } from 'react-bootstrap';
import { connect } from 'react-redux';
import * as PassengerAction from '../../../redux/actions/passengerAction';

import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import './passenger.scss';
import PropTypes from 'prop-types';

export const PassengerFilter = ({ currentUser, filterByFormField }) => {
  const inFlight = window.location.pathname.includes('in-flight');

  return (
    <>
      <Container className="passenger-filter-container">
        <Row>
          <FormGroup row>
            {currentUser && currentUser.role === 'admin' && (
              <>
                <FormControlLabel
                  control={<Checkbox value="passport" />}
                  label="Filter by passport"
                  onChange={filterByFormField}
                />
                <FormControlLabel
                  control={<Checkbox value="address" color="primary" />}
                  label="Filter by address"
                  onChange={filterByFormField}
                />
                <FormControlLabel
                  control={<Checkbox value="dob" color="primary" />}
                  label="Filter by date of birth"
                  onChange={filterByFormField}
                />
              </>
            )}

            {currentUser && currentUser.role === 'staff' && !inFlight && (
              <>
                <FormControlLabel
                  control={<Checkbox value="checkeIn" color="primary" />}
                  label="Filter by Check in "
                  onChange={filterByFormField}
                />
                <FormControlLabel
                  control={<Checkbox value="unCheck" color="primary" />}
                  label="Filter by Uncheck in"
                  onChange={filterByFormField}
                />
                <FormControlLabel
                  control={<Checkbox value="wheelChair" color="primary" />}
                  label="Filter by wheel chair"
                  onChange={filterByFormField}
                />
                <FormControlLabel
                  control={<Checkbox value="infant" color="primary" />}
                  label="Filter by infant"
                  onChange={filterByFormField}
                />
              </>
            )}
          </FormGroup>
        </Row>
      </Container>
    </>
  );
};

const mapDispatchToProps = {
  filterPassenger: PassengerAction.filterPassenger
};
export default connect(null, mapDispatchToProps)(PassengerFilter);
PassengerFilter.propTypes = {
  currentUser: PropTypes.object,
  filterByFormField: PropTypes.func
};
