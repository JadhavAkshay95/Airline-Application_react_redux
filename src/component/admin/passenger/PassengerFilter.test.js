/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import { PassengerFilter } from './passengerFilter'

const renderPassengerFilter = args => {
  const defaultProps = {
    filterPassenger: {},
    currentUser: {},
    filterByFormField: () => {}
  }

  const props = { ...defaultProps, ...args }
  return shallow(<PassengerFilter {...props} />)
}

describe('Passenger filter component', () => {
  it('Should show filters based ', () => {
    // @ts-ignore
    global.window = { location: { pathname: '/admin' } }
    const currentUser = { role: 'admin' }
    const wrapper = renderPassengerFilter({ currentUser })
    expect(wrapper).toMatchSnapshot()
  })
})
