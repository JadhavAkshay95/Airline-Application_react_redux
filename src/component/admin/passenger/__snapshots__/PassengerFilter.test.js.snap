// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passenger filter component Should show filters based  1`] = `ShallowWrapper {}`;
